### PR TITLE
Create form_for for rails

### PR DESCRIPTION
**Pull request by AlanSoto31**
In this project we created a simple form in order to understand how to manage the communication between the different sections of the MVC, this includes:

Models, Views, and Controllers (MVC)
Data Structures & Relationships
Migrations
Routing
Associations
Validations
RESTful design

### CODE REVIEW

File: config/routes.rb
Comment (ghost) [inline config/routes.rb]: Use the` :only` option to generate the routes needed for your resource. See [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_route1.png) for new user routes and see [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_route2.png) for edit user routes.

File: app/controllers/users_controller.rb
Comment (ghost) [inline app/controllers/users_controller.rb]: These two actions will be redundant when you use the` :only` option in your `config/routes.rb`.

File: app/controllers/users_controller.rb
Comment (ghost) [inline app/controllers/users_controller.rb]: Remove the commented code on line 16.

File: app/views/users/new.html.erb
Comment (ghost) [inline app/views/users/new.html.erb]: Line 14 should be `email_field_tag(:email)`

File: app/views/users/new.html.erb
Comment (ghost) [inline app/views/users/new.html.erb]: Line 16 should be` password_field_tag(:password)`.

File: app/views/users/new.html.erb
Comment (ghost) [inline app/views/users/new.html.erb]: Line 27 should be `type='email'`.

File: app/views/users/new.html.erb
Comment (ghost) [inline app/views/users/new.html.erb]: Line 29 should be `type='password'`.

File: app/views/users/new.html.erb
Comment (ghost) [inline app/views/users/new.html.erb]: Line 4 should be `f.email_field` and line 5 should `f.password_field`

Make the above corrections in the` edit.html.erb` file as well.

File: app/views/users/edit.html.erb
Comment (ghost) [inline app/views/users/edit.html.erb]: You should not have any logic in views. If you choose to still use the above logic, then encapsulate in View helpers `(app/helpers/*_helper.rb)`.

**Review by ghost — CHANGES_REQUESTED** [review]
# Status: Changes Required

##  Hello team 😄.

**Great job on your project 💪. However, there are a few fixes needed.**

1). **Readme**

- Add the command for running migrations in your readme instructions manual.

- Explain, in your instructions manual, what routes the user should visit so they can be able to add and edit a user.

2). **Set up `.rubocop.yml` file for rubocop errors. Use this [guideline](https://github.com/microverseinc/linters-config/tree/master/ruby) from Microverse.**

3). **The `views/user/edit.html.erb` should have two commented out forms as well. Remember to update the correct field tags.**

4). **Add `.browserslistrc` in `.gitignore` file and delete it from your repo.**

5). **Check more comments below.**

## Happy coding 😃.

**PS: You can reach me on slack (Rose Wanjohi).**

File: app/views/users/edit.html.erb
Comment (tirthajyoti-ghosh) [inline app/views/users/edit.html.erb]: Please remove this logic from the view and put it in a helper, as the previous reviewer instructed.

That's it!

**Review by tirthajyoti-ghosh — CHANGES_REQUESTED** [review]
# Changes required 🛑 

## Impressive progress so far 💪! Thanks for making all the changes 👏.

### However, we need to work on some changes before we can merge.

You missed one change from the previous review:

- There is still some logic left in `app/views/users/edit.html.erb`. I have pointed out in the comment below.

Your project looks absolutely fantastic. Just make these changes and apply for another code review.

Happy coding 👨‍💻 

*Need help? Shoot me a message on Slack - [@Ghosh](https://app.slack.com/client/T47CT8XPG/DRDM6LAQN/user_profile/URB1MHXK6)*
<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

File: config/routes.rb
Comment (teekaytech) [inline config/routes.rb]: Remove this commented line of code.

File: app/views/users/new.html.erb
Comment (teekaytech) [inline app/views/users/new.html.erb:31]: Since you no longer use these lines of code, kindly remove them.

File: app/views/users/edit.html.erb
Comment (teekaytech) [inline app/views/users/edit.html.erb:36]: Leaving commented lines of code is a bad practice. Kindly remove them.

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've made excellent progress in this project.
However, there's still one more change required before this project can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [ ] It’s better to delete **commented piece of codes**. 
- If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))  
- You are having a lot of commented pieces of code in your project. Specifically, here's the difference between comments about code and commented pieces of code:
 - :x: **commented piece of codes** `<!-- <%= submit_tag("Submit") %> -->`  
 - :heavy_check_mark:  **comments about code** `/* Submit button */`     
- Kindly check the comments below and try to delete all the commented pieces of code as applicable.

* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Comment by teekaytech** [comment]
> ## Status: Changes Required  
> ### Hello team 
> **You've made excellent progress in this project. However, there's still one more change required before this project can be approved. Please check my comments below and fix them as appropriate.**
> 
> **[ REQUIRED CHANGES ]**
> 
> * [x]  It’s better to delete **commented piece of codes**.
> * If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
> * You are having a lot of commented pieces of code in your project. Specifically, here's the difference between comments about code and commented pieces of code:
> * x **commented piece of codes** `<!-- <%= submit_tag("Submit") %> -->`
> * heavy_check_mark  **comments about code** `/* Submit button */`
> * Kindly check the comments below and try to delete all the commented pieces of code as applicable.
> 
> **Please make the required changes and submit another review request when you are done.**
> 
> **Taofeek Olalere [Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
> **Happy Coding smiley**

**Hello team,**

I just realized that leaving the commented `form` and `form_tag` code is a requirement for this project. Kindly ignore my previous comments. Just remove the affected part (i.e. routes) and re-submit for approval. Apologies for any inconvenience.

_Happy coding_

**Review by Dandush03 — APPROVED** [review]
# Status: Project Approved :rocket: 
## Keep Up The Good Work :muscle: 
# Happy Coding :muscle: :computer:

**Pull request merged**
